awscli/customizations/s3/filegenerator.py

```diff
@@ -208,13 +207,19 @@ def list_files(self, path, dir_op):
                         for x in self.list_files(file_path, dir_op):
                             yield x
                     else:
-                        size, last_update = get_file_stat(file_path)
-                        last_update = self._validate_update_time(
-                            last_update, path)
-                        yield (
-                            file_path,
-                            {'Size': size, 'LastModified': last_update}
-                        )
+                        stats = self._safely_get_file_stats(path, file_name=name)
```

Comment:
Oh... I see why you are doing that now! So the line you pointed to in the original is actually a bug...

In that line snippet we are passing ``path`` which is directory to the warning message, when we should be passing the ``file_path`` which is the actual filename that produced the ``last_update`` value. I guess it never went noticed because all it affects is the warning message in that it will print out the directory instead of filename in the warning.

I am all for fixing this to use ``file_path``. It would also simplify your code.